packages/Kokkos/test/tstPointCloud.cpp

```diff
@@ -0,0 +1,196 @@
+  names of its contributors may be used to endorse or promote products
+  derived from this software without specific prior written permission.
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+  A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+  HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+  SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+  LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+  DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+  THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+*/
+//---------------------------------------------------------------------------//
+/*!
+ * \file   tstPointCloud.cpp
+ * \author Stuart Slattery
+ * \brief  Point Cloud unit tests.
+ */
+//---------------------------------------------------------------------------//
+
+#include <DTK_PointCloud.hpp>
+
+#include <Teuchos_CommHelpers.hpp>
+#include <Teuchos_DefaultComm.hpp>
+#include <Teuchos_ScalarTraits.hpp>
+#include <Teuchos_UnitTestHarness.hpp>
+
+#include <Kokkos_Core.hpp>
+#include <Kokkos_Random.hpp>
+
+#include <limits>
+
+//---------------------------------------------------------------------------//
+// FUNCTOR HELPERS
+//---------------------------------------------------------------------------//
+//! Create global ids.
+template <class SC, class LO, class GO, class NO>
+class CreateIdsFunctor
+{
+  public:
+    using PointCloud = DataTransferKit::PointCloud<SC, LO, GO, NO>;
+    using DeviceType = typename NO::device_type;
+    using GlobalIdView = Kokkos::View<GO *, DeviceType>;
+
+    CreateIdsFunctor( GlobalIdView global_ids, int comm_rank )
+        : _global_ids( global_ids )
+        , _comm_rank( comm_rank )
+    { /* ... */
+    }
+
+    KOKKOS_INLINE_FUNCTION
+    void operator()( const size_t p ) const
+    {
+        _global_ids( p ) = _comm_rank * _global_ids.extent( 0 ) + p;
+    }
+
+  private:
+    GlobalIdView _global_ids;
+    int _comm_rank;
+};
+
+//---------------------------------------------------------------------------//
+// TEST TEMPLATE DECLARATIONS
+//---------------------------------------------------------------------------//
+// Point cloud test.
+TEUCHOS_UNIT_TEST_TEMPLATE_4_DECL( PointCloud, basic, SC, LO, GO, NO )
+{
+    // Test types.
+    using PointCloud = DataTransferKit::PointCloud<SC, LO, GO, NO>;
+    using DeviceType = typename NO::device_type;
+    using GlobalIdView = Kokkos::View<GO *, DeviceType>;
+    using CoordinateView = Kokkos::View<SC **, DeviceType>;
+    using ExecutionSpace = typename PointCloud::execution_space;
```

Comment:
Why `ExecutionSpace` and not `execution_space`? Is the implicit agreement that we only use `execution_space` as a typedef in classes?